lib/open-uri.rb

```diff
@@ -148,7 +149,11 @@ def OpenURI.open_uri(name, *rest) # :nodoc:
       end
       encoding = Encoding.find(options[:encoding])
     end
-
+    if options.has_key? :request_specific_fields
+      if !(options[:request_specific_fields].is_a?(Hash) || options[:request_specific_fields].is_a?(Proc))
+        raise ArgumentError, "Invalid request_specific_fields' format: #{options[:request_specific_fields]}"
```

Comment:
I think "format" is not an appropriate word here.
Also, `inspect` should be called for the option value to show it for users.

So, how about making the exception message
"Invalid request_specific_fields: #{options[:request_specific_fields].inspect}"?

Reply:
Thanks you so much. I completely agree with your suggestion. Adding inspect will indeed provide users with more detailed information when debugging.
I also added the word option to the message to maintain consistency with other option-related error messages.